frontend/src/app/main/ui/measurements.cljs

```diff
@@ -475,25 +468,469 @@
+                h (- (:y2 block) (:y1 block))]
+            (create-column-gap-display-data x y h)))
+
+        create-cgdd-block
+        (fn [block]
+          (let [x (if flip-x
+                    (- (:x1 block) (:column-gap gap))
+                    (:x2 block))
+                y (if flip-y
+                    (+ y1 (:p1 padding))
+                    (+ y1 (:p3 padding)))
+                h (- height (+ (:p1 padding) (:p3 padding)))]
+            (create-column-gap-display-data x y h)))
+
+        create-row-gap-display-data
+        (fn [x y width]
+          {:frame-id frame-id
+           :zoom zoom
+           :gap-type :row-gap
+           :gap gap
+           :on-mouse-enter (partial on-mouse-enter :row-gap (:row-gap gap))
+           :on-mouse-leave on-mouse-leave
+           :rect-data {:x x
+                       :y y
+                       :width width
+                       :height (:row-gap gap)
+                       :initial-value (:row-gap gap)
+                       :resize-type :bottom
+                       :resize-axis :y
+                       :resize-negate? (:row-gap negate)}
+           :hover?    (= @hover :row-gap)
+           :selected? (= gap-selected :row-gap)
+           :mouse-pos mouse-pos})
+
+        create-rgdd
+        (fn [shape]
+          (let [block  (get-container-block shape)
+                x (:x1 block)
+                y (if flip-y
+                    (- (:y1 (:selrect shape))
+                       (get-in shape [:layout-item-margin :m3])
+                       (:row-gap gap))
+                    (+ (:y2 (:selrect shape)) (get-in shape [:layout-item-margin :m3])))
+                w (- (:x2 block) (:x1 block))]
+            (create-row-gap-display-data x y w)))
+
+        create-rgdd-block
+        (fn [block]
+          (let [x (if flip-x
+                    (+ x1 (:p2 padding))
+                    (+ x1 (:p4 padding)))
+                y (if flip-y
+                    (- (:y1 block) (:row-gap gap))
+                    (:y2 block))
+                w (- width (+ (:p2 padding) (:p4 padding)))]
+            (create-row-gap-display-data x y w)))
+
+        pill-width                 (/ flex-display-pill-width zoom)
+        pill-height                (/ flex-display-pill-height zoom)
+        mouse-pill-data            {:height pill-height
+                                    :width pill-width
+                                    :font-size (/ font-size zoom)
+                                    :border-radius (/ flex-display-pill-border-radius zoom)
+                                    :color distance-color
+                                    :x (:x @mouse-pos)
+                                    :y (- (:y @mouse-pos) pill-width)
+                                    :value @hover-value}]
+
+    [:g.gaps {:pointer-events "visible"}
+     [:*
+      (for [block (drop-last wrap-blocks)]
+        (if is-col?
+          [:& gap-display (create-cgdd-block block)]
```

Comment:
This is not a good idea. Props should be static maps or javascript objects. The only thing that changes its the rect-data and the rest of the information should be in the component call. If you want to reuse de properties you need to do it in a different way.